docs/api-spec-swagger.yaml

```diff
@@ -0,0 +1,3370 @@
+  #     description: Returns detailed analytics data for a specific job.Must have ADMIN authority.
+  #     security: 
+  #     - jwt: []
+  #     operationId: getJobAnalytics
+  #     parameters:
+  #       - name: jobId
+  #         in: path
+  #         description: Id of the job to get analytics for
+  #         required: true
+  #         schema:
+  #           type: integer
+  #           format: int64
+  #       - name: startDate
+  #         in: query
+  #         description: Start date for analytics (format YYYY-MM-DD)
+  #         schema:
+  #           type: string
+  #           format: date
+  #       - name: endDate
+  #         in: query
+  #         description: End date for analytics (format YYYY-MM-DD)
+  #         schema:
+  #           type: string
+  #           format: date
+  #     responses:
+  #       '200':
+  #         description: Successful operation
+  #         content:
+  #           application/json:
+  #             schema:
+  #               $ref: '#/components/schemas/JobViews'
+  #       '400':
+  #         description: Invalid Id supplied
+  #       '404':
+  #         description: Job not found      
+components:
+  schemas:
+    Job:
+      type: object
+      required:
+        - title
+        - level_id
+        - category_id
+      properties:
+        job_id:
+          type: string
+          format: uuid
+          example: f21e77c9-8fd6-4d58-ae91-2560b53c2389
+        title:
+          type: string
+          example: Software Engineer
+        level_id:
+          type: string
+          format: uuid
+          example: f21e77c9-8fd6-4d58-ae91-2560b53c2389
+        category_id:
+          type: string
+          format: uuid
+          example: f21e77c9-8fd6-4d58-ae91-2560b53c2389
+        location:
+          type: string
+          example: San Francisco, CA
+        description:
+          type: string
+          example: We are looking for a skilled software engineer...
+        compensation:
+          type: string
+          example: $120,000 - $150,000 per year
+        application_link:
+          type: string
+          example: https://apply.jobboard.com/job/10
+        is_external:
+          type: boolean
+          example: false
+        company_id:
+          type: string
+          format: uuid
+          example: f21e77c9-8fd6-4d58-ae91-2560b53c2389
+        created_at:
+          type: string
+          format: date-time
+        updated_at:
+          type: string
+          format: date-time
+        tags:
+          type: string
+          format: text
+    JobCategory:
+      type: object
+      properties:
+        id:
+          type: string
+          format: uuid
+          example: f21e77c9-8fd6-4d58-ae91-2560b53c2389
+        title:
+          type: string
+          example: Technology
+    JobLevel:
+      type: object
+      properties:
+        id:
+          type: string
+          format: uuid
+          example: f21e77c9-8fd6-4d58-ae91-2560b53c2389

```

Comment:
minor comment the id here is int. Same for Category and type and all non-major entities. major entities are (job, event, resource, company, user).